src/client.rs

```diff
@@ -236,18 +238,41 @@ impl Client {
 
         add_invite_event_handler(client, tx, &user_id).await;
         add_room_message_event_handler(client, tx, &user_id, &self.event_notifier).await;
+        add_created_room_event_handler(client, user_notifier, tx).await;
 
         let (cancel_sync, check_cancel) = async_channel::bounded::<bool>(1);
 
         tokio::spawn(sync_until_cancel(client, check_cancel).await);
 
         let res = response.expect("already checked it is not an error");
-        let joined_rooms = res.rooms.join.keys().cloned().collect::<Vec<_>>();
+        let mut direct_messages = res.rooms.join.keys().cloned().collect::<Vec<_>>(); // here joined rooms but after cleaning up, it'll be just direct messages (private rooms)
         let invited_rooms = res.rooms.invite.keys().cloned().collect::<Vec<_>>();
+        let mut channels: Vec<OwnedRoomId> = Vec::new();
+        let mut indexes: Vec<usize> = Vec::new(); // collect indexes to remove from direct_messages
+
+        for (i, id) in direct_messages.iter().enumerate() {
+            match client.get_room(id) {
+                Some(room) => {
+                    if is_channel(&room) {
+                        channels.push(id.to_owned());
+                        indexes.push(i);
+                    }
+                }
+                None => log::debug!("room not found in store {}", id),
+            }
+        }
+
+        // remove public rooms from direct messages (private rooms)
+        for i in indexes {
+            log::debug!("removing {i} from joined rooms");
+            direct_messages.remove(i);
+        }
+
```

Comment:
We can avoid `indexes` and be more clear if we start with two empty lists for `direct_messages` and `channels`, then fill them with `res.rooms.join` rooms depending on the case.